src/tree_types.ml

```diff
@@ -0,0 +1,265 @@
+(** Miscellaneous types and utilities for decision trees. *)
+open Extra
+
+(** {3 Tree constructors} *)
+
+(** Define a simpler representation of terms. *)
+
+(** A constructor is the representation of a symbol along with the number of
+    arguments to which it is applied. *)
+type treecons =
+  | Symb of
+      { c_mod : Files.module_path
```

Comment:
I have mixed feelings about using the record syntax here: we don't do that anywhere else, and since all the components live in different types it is not really a problem to use a tuple instead.